sympy/series/order.py

```diff
@@ -212,6 +212,8 @@ def __new__(cls, expr, *args, **kwargs):
                 # x*y (wrong order term!).  That's why we want to deal with
                 # expand()'ed expr (handled in "if expr.is_Add" branch below).
                 expr = expr.expand()
+            else:
+                expr = expr.factor()
```

Comment:
It seems that this aims to transform the sum  `1/x - 3/(3*x + 2)` into a product, so it looks like it should only be applied to instances of `Add`. The code would probably be better placed here: https://github.com/sympy/sympy/blob/f7318dbfef2bf61e5ed8bc55f3e374880ec6ea67/sympy/series/order.py#L199-L201

It is possible that `expand_multinomial` could be removed and replaced by this. You could also test something like
```
                num, den = expr.as_numer_denom()
                expr = num/den
```
instead of `expr.factor()` which may be more expensive.

Reply:
> It seems that this aims to transform the sum 1/x - 3/(3*x + 2) into a product, so it looks like it should only be applied to instances of Add. The code would probably be better placed here:
> 
> sympy/sympy/series/order.py
> 
> Lines 199 to 201 in f7318db
>  if expr.is_Add: 
>      from sympy import expand_multinomial 
>      expr = expand_multinomial(expr) 

Right. I changed the location of `factor`.

> It is possible that expand_multinomial could be removed

Yes, I run the tests and they pass also without `expand_multinomial`.

> You could also test something like
> 
>                 num, den = expr.as_numer_denom()
>                 expr = num/den
> 
> instead of expr.factor() which may be more expensive.
> 

Putting this instead of `expr.factor()` causes 2 failures in tests. The tests which failure are:

- `test_wester.py` at lines 2088-2090.
- `test_trigonometric.py` at llines 1224-1225.

Reply:
> Putting this instead of `expr.factor()` causes 2 failures in tests. The tests which failure are:
> 
> * `test_wester.py` at lines 2088-2090.
> * `test_trigonometric.py` at llines 1224-1225.

Can you tell what is `expr` in those cases when using `num/den` and when you use `expr.factor`?

Reply:
For `test_wester`.

```
expr = -exp(-_p**2/(_p + 1) + _p)*exp(_p**2/(_p + 1) - _p)/2 - exp(-_p**2/(_p + 1) + _p)*exp(_p**2/(_p + 1) - _p)/(2*_p)
expr.factor() = -(_p + 1)/(2*_p)
num = -_p*exp(-_p**2/(_p + 1) + _p)*exp(_p**2/(_p + 1) - _p) - exp(-_p**2/(_p + 1) + _p)*exp(_p**2/(_p + 1) - _p)
den = 2*_p
```
In this case the test does  not give an error, but it never finishes.

For `test_trigonometric`.
```
expr = -4*_Dummy_69 + (_Dummy_69 + 1)**4 - 4*(_Dummy_69 + 1)**3 + 6*(_Dummy_69 + 1)**2 - 3
expr.factor() = _Dummy_69**4
num = -4*_Dummy_69 + (_Dummy_69 + 1)**4 - 4*(_Dummy_69 + 1)**3 + 6*(_Dummy_69 + 1)**2 - 3
den = 1
```

Reply:
It looks like only `factor` factor will help here.